nemo/collections/asr/parts/preprocessing/segment.py

```diff
@@ -241,7 +241,7 @@ def from_file(
                     sample_rate = f.samplerate
                     if offset > 0:
                         f.seek(int(offset * sample_rate))
-                    if duration > 0:
+                    if duration is not None and duration > 0:
```

Comment:
This is accepted change however I am curious if it did raise `None cannot be compared with integer` error? I use duration as null for all evaluations with speaker models and they run fine?

Reply:
hmm, interesting....at least with python3.10 it raised the error when I tried